### PR TITLE
Clean-up in pointer logic

### PR DESCRIPTION
**Pull request by romainbrenguier**
No functional changes.
Remove useless definitions and make documentation be doxygen.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: f1b2f7cbdcc946a675f12cddd0b7815a93be8d17).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/116263182

File: src/solvers/flattening/pointer_logic.h [hunk 1]
```diff
@@ -40,12 +39,12 @@ class pointer_logict
     }
   };
 
-  // converts an (object,offset) pair to an expression
+  /// converts an (object,offset) pair to an expression
```
Comment (peterschrammel) [inline src/solvers/flattening/pointer_logic.h]: Convert

File: src/solvers/flattening/pointer_logic.h [hunk 2]
```diff
@@ -40,12 +39,12 @@ class pointer_logict
...
   exprt pointer_expr(
     const pointert &pointer,
     const pointer_typet &type) const;
 
-  // converts an (object,0) pair to an expression
+  /// converts an (object,0) pair to an expression
```
Comment (peterschrammel) [inline src/solvers/flattening/pointer_logic.h]: Convert

File: src/solvers/flattening/pointer_logic.h [hunk 3]
```diff
@@ -55,13 +54,13 @@ class pointer_logict
...
 
-  // number of NULL object
+  /// number of NULL object
   std::size_t get_null_object() const
   {
     return null_object;
   }
 
-  // number of INVALID object
+  /// number of INVALID object
```
Comment (peterschrammel) [inline src/solvers/flattening/pointer_logic.h]: Return number...

**Review by peterschrammel — APPROVED** [review]

**Review by forejtv — APPROVED** [review]

**Review by kroening — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: bb4a264f15236c3dbc063f5dcb322f58a5c969e1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/116293832

**Pull request merged**